torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -175,11 +177,15 @@ void RRefContext::delUser(
       // Sending an RRefUserDelete causes the receiver to run delForkOfOwner,
       // which is now idempotent. See the comment at RRefContext::delForkOfOwner
       // for more details.
+      ++numPendingFutures_;
```

Comment:
This looks OK to me. I wonder would it be better if we make this a guard with a move ctor?

Reply:
As you prefer. We could indeed create a small type, which increases this count in the constructor and decreases it in the destructor, and capture an object of that class in the closure.

Reply:
no need to change in this PR. And honestly there is no clear winner which one is better.